cmd/kubeapps-apis/plugins/fluxv2/packages/v1alpha1/server.go

```diff
@@ -238,14 +258,21 @@ func (s *Server) pullChartTarball(ctx context.Context, packageRef *corev1.Availa
 		Resource: fluxHelmCharts,
 	}
 
+	// TODO: use the namespace from the request
 	resourceIfc := client.Resource(chartsResource).Namespace("default")
 
 	// see if we the chart already exists
-	// TODO You should be able to use the metav1.ListOptions{} above to specify
-	// filtering using the FieldSelector. More info at
-	// https://kubernetes.io/docs/concepts/overview/working-with-objects/field-selectors/.
+	// TODO:
+	// https://github.com/kubeapps/kubeapps/pull/2915
+	// @minelson says: You should be able to use the metav1.ListOptions{} above
+	// to specify filtering using the FieldSelector.
+	// More info at https://kubernetes.io/docs/concepts/overview/working-with-objects/field-selectors/ .
 	// You may even be able to specify that the pull should be complete to be included (ie.
 	// that status conditions ready is true), not sure, but that'd be nice.
+	// @gfichtenholt As I remember I did try a few things here. The problem is the set of supported
+	// fields in FieldSelector is very small. things like spec.chart are certainly not supported.
+	// see kubernetes/client-go#713 and https://github.com/flant/shell-operator/blob/8fa3c3b8cfeb1ddb37b070b7a871561fdffe788b/// HOOKS.md#fieldselector. Nevertheless, I made a TODO comment here just as you suggested to see
+	// if I can improve later
```

Comment:
Personally don't think we need the history of the discussion there, I think it's enough to just state the current state - something along the lines of the ListOptions FieldSelector not supporting the fields required (and link to the relevant issue/pr comment for more context, if it's helpful).

Reply:
ok you got it